### PR TITLE
transcribe: alert the user when a page is being transcribed

### PR DESCRIPTION
**Pull request by diegoviola**

### CODE REVIEW

**Comment by benwbrum** [comment]
When I run this locally, I do not see the warning message.  Instead, I see this error in the Javascript console:
```
:3000/benwbrum/1/17/still_editing/6086:1 Failed to load resource: net::ERR_CONNECTION_REFUSED
```

**Comment by benwbrum** [comment]
The error message in the javascript console appears to be unrelated.  I can no longer duplicate it.

File: app/controllers/transcribe_controller.rb [hunk 1]
```diff
@@ -29,6 +29,7 @@ def display_page
     @auto_fullscreen = cookies[:auto_fullscreen] || 'no';
     @layout_mode = cookies[:transcribe_layout_mode] || @collection.default_orientation
     session[:col_id] = @collection.slug
+    flash[:alert] = "This page is being edited by another user!" if @page.edit_started_at > Time.now + 1.minute unless @page.edit_started_at.nil?
```
Comment (benwbrum) [inline app/controllers/transcribe_controller.rb]: This needs to be changed from `Time.now + 1.minute` to `Time.now - 1.minute`.

**Comment by diegoviola** [comment]
@benwbrum I pushed a change that introduces the `edit_started_by_user` that you suggested in #1063 

Now if I click on the same transcribe link from another browser I get the warning as it's supposed to happen (with a different user).

Thanks for your help with this.

**Comment by benwbrum** [comment]
I've tested this and found an odd edge case. 

* User A starts editing page 1
* User B clicks the transcribe tab on page 2
  * User B sees the warning message
  * User B's browser calls the `still_editing` route (unknown to them), updating the edit time and user_id
* User B clicks to page 2
* User B clicks back to page 1
* User B does _not_ see the warning message, even though user A is still editing the page

I _think_ this can be solved by stopping the Javascript from being called if the warning message was displayed.  I'm not sure whether it's better to do this client-side (by looking at the `flash`), or server-side (by looking at `@page.edit_started_by_user_id` and the time window).  It might be better to look in the flash since that keeps us from duplicating the time window logic.

**Pull request merged**